tests/sbc_test.py

```diff
@@ -165,63 +225,145 @@ def test_sbc_accuracy():
-    assert (checks["c2st_ranks"] < 0.55).all()
-    assert (checks["c2st_dap"] < 0.55).all()
+
+    # With artificial uniform ranks, test statistics should indicate uniformity
+    assert (checks["ks_pvals"] > 0.05).all(), "KS test failed on uniform ranks"
+    assert (checks["c2st_ranks"] < 0.55).all(), "C2ST failed on uniform ranks"
+    assert (checks["c2st_dap"] < 0.55).all(), (
+        "C2ST failed on prior-distributed DAP samples"
+    )
 
 
-# add test for sbc plotting
 @pytest.mark.parametrize("num_bins", (None, 30))
 @pytest.mark.parametrize("plot_type", ("cdf", "hist"))
 @pytest.mark.parametrize("legend_kwargs", (None, {"loc": "upper left"}))
 @pytest.mark.parametrize("num_rank_sets", (1, 2))
 def test_sbc_plotting(
-    num_bins: int, plot_type: str, legend_kwargs: Union[None, dict], num_rank_sets: int
+    num_bins: Optional[int],
+    plot_type: str,
+    legend_kwargs: Optional[Dict],
+    num_rank_sets: int,
 ):
-    """Test the uniformity checks for SBC."""
-
+    """Test SBC plotting functionality with various options."""
     num_dim = 2
     num_posterior_samples = 1000
 
-    # Ranks should be distributed uniformly in [0, num_posterior_samples]
+    # Generate artificial uniform ranks for testing visualization
     ranks = [
         torch.distributions.Uniform(
             zeros(num_dim), num_posterior_samples * ones(num_dim)
         ).sample((num_posterior_samples,))
     ] * num_rank_sets
 
-    sbc_rank_plot(
+    # Test that plotting function runs without errors
+    fig = sbc_rank_plot(
         ranks,
         num_posterior_samples,
         num_bins=num_bins,
         plot_type=plot_type,
         legend_kwargs=legend_kwargs,
     )
+
+    # Basic check that figure was created
+    assert fig is not None, "Plot function should return a figure"
+
+
+@pytest.mark.parametrize("num_workers", [1, 2])
+def test_sbc_parallelization(num_workers: int, gaussian_setup: Dict):
+    """Test that SBC produces consistent results with different worker counts."""
+    prior = gaussian_setup["prior"]
+    simulator = gaussian_setup["simulator"]
+
+    # Parameters
+    num_simulations = 200
+    num_sbc_runs = 10
+    num_posterior_samples = 50
+
+    # Train model
+    posterior = train_inference_method(
+        NPE, prior, simulator, num_simulations=num_simulations, max_num_epochs=2
+    )
+
+    # Generate test data
+    thetas = prior.sample((num_sbc_runs,))
+    xs = simulator(thetas)
+
+    # Run SBC with specified number of workers
+    ranks, _ = run_sbc(
+        thetas,
+        xs,
+        posterior,
+        num_posterior_samples=num_posterior_samples,
+        num_workers=num_workers,
+    )
+
+    # Check shape
+    assert ranks.shape == (num_sbc_runs, gaussian_setup["num_dim"]), (
+        f"Ranks shape incorrect for {num_workers} workers"
+    )
+
+
+@pytest.mark.parametrize("batch_sampling", [True, False])
+def test_sbc_batch_sampling(batch_sampling: bool, gaussian_setup: Dict):
```

Comment:
lots of new tests. What is their runtime? Should they be marked as slow or could they be merged with other tests (in the spirit of keeping test time low)?

Reply:
good point, but due to the new fixtures it's only 4 * 0,03s more than before.